ioc/credentials.cpp

```diff
@@ -24,9 +24,26 @@ namespace ioc {
  */
 
 Credentials::Credentials(const server::ClientCredentials& clientCredentials) {
-    // Extract host name part (or whole thing if no colon present)
-    auto pos = clientCredentials.peer.find_first_of(':');
-    host = clientCredentials.peer.substr(0, pos);
+    // There are 3 possibilities of addresses
+    // IPv4 wrapped as IPV6 : e.g.: [::ffff:192.168.1.1]:9999
+    // IPv6: e.g. [2001:0db8:85a3:0000:0000:8a2e:0370:7334]:9999
+    // IPv4 only: e.g.: 192.168.1.1:9999
+    auto pos = clientCredentials.peer.find_first_of("[::ffff:");
+    auto end = clientCredentials.peer.find_first_of(']');
+    //IPv4 wrapped as IPv6
+    if (pos != clientCredentials.peer.npos) {
+        //len("[::ffff:") == 8
+        host = clientCredentials.peer.substr(8, end - 8);
+    }
+    //IPv6
+    else if (end != clientCredentials.peer.npos)
+        host = clientCredentials.peer.substr(1, end-1);
+    //IPv4
+    else {
+      // Extract host name part (or whole thing if no colon present)
+      pos = clientCredentials.peer.find_first_of(':');
+      host = clientCredentials.peer.substr(0, pos);
+    }
```

Comment:
Did you find `evutil_parse_sockaddr_port()` and `evutil_inet_ntop()`?

Reply:
No, we didn't. Question: 
@mdavidsaver  How do you think that these functions could help us ?

Reply:
`evutil_parse_sockaddr_port()` will parse `<host>:<port>`, considering the various forms of IPv4/6 addresses.  `evutil_inet_ntop()` renders an IPv4/6 address as a string.

As I think about it though, it would be simpler to use `pvxs::SockAddr`, which already knows about IPv6 address encodings.  The printer for that type will omit the port number if it is zero.

```c++
SockAddr addr(clientCredentials.peer);
addr.setPort(0);
std::string host(SB()<<addr);
```

Reply:
So, the solution using `pvxs::SockAddr`  gets the right address, but it seems it cannot handle IPv6 addresses, or at least cannot handle this kind of address:
`[::ffff:192.168.1.1]` or `::ffff:192.168.1.1`
For the current code work it is needed to unwrap the IPv4 

I will try to understand better what is going on for these IPv6 addressing not working

Reply:
As it happens, I think I have a way to handle this from another branch.  I have pushed a commit which adds `SockAddr::map6to4()` to "unwrap" IPv4 addresses encoded as IPv6.

Reply:
Ah great, I tested and it seems to work fine! I changed my commit ut use the map6to4